zio-http/shared/src/main/scala/zio/http/endpoint/openapi/JsonSchema.scala

```diff
@@ -1209,7 +1220,7 @@ object JsonSchema {
 
   final case class Object(
     properties: Map[java.lang.String, JsonSchema],
-    additionalProperties: Either[Boolean, JsonSchema],
+    additionalProperties: Either[Boolean, Object.Dictionary],
```

Comment:
This is a similar discussion as we had today in the other issues. Your focus here is only the code gen. But this is a general impl. of json schema. And it does not know Dictionary.
I think Schema is right and you need to fix your requirements in the gen

Reply:
Indeed.
In fact, after giving it more thought, the extension can take any schema, not just a reference. as long as we validate it is a string schema.
So I'll generalize to also allow inline a regular schema instead of restricting to ref.
Will update :+1:

Reply:
@987Nabil both on write & read side, I'll need some sort of "tuple" to hold the optional key schema with the required value schema.

So I have several options here:
- make JsonSchema.Object take another optional key schema field
- make additionalProperties a tuple `(Option[JsonSchema], Either[Boolean, JsonSchema])` - for optional key schema with current either
- don't touch structure, but use a JsonSchema "product" for the two optional schemas - currently perhaps only `AllOfSchema` makes sense

I don't like the first, since it's not "bound" to where it's eligible, which is the `additionalProperties` field.
The last  would be ambiguous whether top level allOf refer to key & value schema, or just an allOf for the values.
The second option is too verbose.
IIUC, `Left(false)` means no `additionalProperties`. `Left(true)` means unconstrained `additionalProperties` values. And `Right(schema)` means strict schema for values.
If that's the case, it could have been `Option[Option[JsonSchema]]`:
- None == Left(false)
- Some(None) == Left(true)
- Some(Some(schema)) == Right(schema)
Such that with optional key schema it can be `Option[(Option[JsonSchema], Option[JsonSchema])]` - the tuple is optional key schema + optional value schema.
But if that's the case, why not make it a named tuple (case class) like `Dictionary`?

I would make sure to do both read/write sides to `SerializableJsonSchema` work with it, but it does seem like the best option. no?

Reply:
Leave additional properties as is. When you read the key extension, add a new annotation with the schema to the schema of additional properties. Less complicated. And fits the idea of it being an extension better.

Reply:
@987Nabil annotate how?
The current annotations are: Examples,  Default,  Discriminator,  Nullable,  Description,  ContentEncoding,  ContentMediaType,  Deprecated
Should I just add a new one?
Also, in case we have `Left(true)` together with my extension, i.e. only restricting keys with schema, not values.
I have nothing to annotate…

Reply:
Yes, just add a new one. Is true even a valid map? I think true is just a way to allow extra fields (instead of failing validation) or for dynamic payload.